src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -28,10 +29,31 @@
 from sagemaker_xgboost_container.callback import add_debugging
 from sagemaker_xgboost_container.constants.xgb_constants import CUSTOMER_ERRORS
 
+MODEL_NAME = "xgboost-model"
 
 logger = logging.getLogger(__name__)
 
 
+def add_sigterm_handler(model_dir, is_master):
+    """Stop training and cleanup model directory when SIGTERM is received.
+
+    Model directory is only cleaned if is_master is True. Otherwise program terminates.
+
+    :param model_dir: Directory where model is saved
+    :param is_master: True if single node training, or the current node is the master node in distributed training
+    """
+    def _terminate():
+        os._exit(0)
```

Comment:
os._exit(0) is used since it is an Async-signal safe function, while sys.exit(0) is not. Unsafe functions can sometimes not work during interruption of other unsafe calls such as system calls.